sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -588,14 +588,19 @@ object FunctionRegistry {
           val validParametersCount = constructors
             .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
             .map(_.getParameterCount).distinct.sorted
-          val expectedNumberOfParameters = if (validParametersCount.length == 1) {
-            validParametersCount.head.toString
-          } else {
-            validParametersCount.init.mkString("one of ", ", ", " and ") +
-              validParametersCount.last
+          val expectedErrorMsg = validParametersCount.length match {
+            case 0 =>
+              ""
```

Comment:
nit: in case of 0, I'd prefer a more generic `Invalid arguments for function $name`, instead of `Invalid number of arguments for function $name`. Because as I mentioned earlier, I think it is questionable which is the number of arguments in this case...

Reply:
Done